src/apiGatewayTest/java/uk/gov/hmcts/reform/blobrouter/ApiGatewayTest.java

```diff
@@ -0,0 +1,172 @@
+package uk.gov.hmcts.reform.blobrouter;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import io.restassured.RestAssured;
+import io.restassured.config.RestAssuredConfig;
+import io.restassured.config.SSLConfig;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ApiGatewayTest {
+
+    private static final String SUBSCRIPTION_KEY_HEADER_NAME = "Ocp-Apim-Subscription-Key";
+    private static final String PASSWORD_FOR_UNRECOGNISED_CLIENT_CERT = "testcert";
+
+    private static final Config CONFIG = ConfigFactory.load();
+    private static final String API_URL = CONFIG.getString("api.gateway-url");
+    private static final String SAS_TOKEN_ENDPOINT = "/token/bulkscan";
+    private static final String SUBSCRIPTION_KEY = CONFIG.getString("client.subscription-key");
+
+    private static File validJavaKeyStore;
+    private static final String VALID_KEY_STORE_PASSWORD = CONFIG.getString("client.valid-key-store.password");
+
+    @BeforeAll
+    static void checkConfig() throws IOException {
+        assertThat(API_URL).as("API gateway URL").isNotEmpty();
+        assertThat(SUBSCRIPTION_KEY).as("Subscription key").isNotEmpty();
+
+        // create tmp file for valid certificate retrieved from key vault
+        validJavaKeyStore = File.createTempFile("appGW", "test");
+
+        try (FileOutputStream fos = new FileOutputStream(validJavaKeyStore)) {
+            fos.write(Base64.getDecoder().decode(CONFIG.getString("client.valid-key-store.content")));
+        }
+    }
+
+    @Test
+    void should_accept_request_with_valid_certificate_and_subscription_key() {
```

Comment:
can we also please add test case for valid cert and invalid sub key ?

Reply:
That's second (the followin ❗️) test case 😑

Reply:
:)